### PR TITLE
Fallback to Fedora29 minimal for images

### LINKED ISSUE #914: qemu-img 3.1 failing when trying to do stream conversions of cirros image
/kind bug

**What happened**:

When attempting to use the CDI importer to create a DataVolume (or a PVC directly) and specifying a cirros image, the conversion process hangs and never completes.

**What you expected to happen**:

The conversion process should either finish or fail, terminating the importer pod and providing me with a PVC that has the expected image on it (upon success).

**How to reproduce it (as minimally and precisely as possible)**:

Create a DV using the follwing yaml:

```
apiVersion: cdi.kubevirt.io/v1alpha1
kind: DataVolume
metadata:
  name: "fedora-datavol"
  annotations:
    kubevirt.io/provisionOnNode: kube-1
spec:
  source:
      http:
         url: "http://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"
  pvc:
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: "1Gi"
```
Then tail the logs on the importer pod and note the hang

### PR DESCRIPTION
**Pull request by j-griffith**
The newer version of qemu-img (3.1.0) that ships in Fedora 30 has issues
with streaming conversions whereby it hangs during the conversion
process.  Rather than try to force a downgrade of qemu when building
images instead, let's try reverting back to Fedora29 minimal which has
qemu-img version  3.0.0.

This issue has also exposed the fact that we aren't performing any
testing against streaming external images, we're only testing against
images that we've pulled in to our test infrastructure.  So add a
functional test that builds a DataVolume directly from the cirros site.

**What this PR does / why we need it**:

It's been discovered that qemu-img version 3.1.0 which ships in Fedora-30 hangs when trying to do a streaming conversion of the external cirros image.  It also turns out that we weren't detecting this via our test automation because we're running in an air gapped env with our own VM images.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #914 

**Special notes for your reviewer**:

**Release note**:

```release-note
Downgrade base images from Fedora30 to Fedora29
```

### CODE REVIEW

**Comment by awels** [comment]
Do we really have to downgrade fedora to 29 for all the containers? The only one that really needs it is the importer container. The rest should be fine.

**Comment by j-griffith** [comment]
> Do we really have to downgrade fedora to 29 for all the containers? The only one that really needs it is the importer container. The rest should be fine.

Well, personally I like consistency and I didn't see any downside to downgrading.  If there's some advantage to using F30 that I'm not thinking of I'm fine either way.  I just wanted to be as consistent and vanilla as possible and NOT have to manage another corner case going forward, we have enough of those already.

File: tests/utils/datavolume.go [hunk 1]
```diff
@@ -36,6 +36,8 @@ const (
 	InvalidQcowImagesURL = "http://cdi-file-host.%s/invalid_qcow_images/"
 	// TarArchiveURL provides a test url for a tar achive file
 	TarArchiveURL = "http://cdi-file-host.%s/archive.tar"
+	// CirrosExternalURL provides the standard cirros image download url external from our test env
+	CirrosExternalURL = "http://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"
```
Comment (awels) [inline tests/utils/datavolume.go]: I don't think I want to depend on an external source. from CDIs perspective there is no difference in going to the internet and going to an http server hosted in the cluster (besides maybe which network to go on). We already have the cirros image in the repo tests/images/cirros.qcow2 we just need to copy it into the http container in the right directory. /tmp/shared/images

Comment (j-griffith) [reply tests/utils/datavolume.go]: No problem, I can change that if you think it's a safe assumption.

File: tests/datavolume_test.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 	tinyCoreIsoRegistryURL := fmt.Sprintf(utils.TinyCoreIsoRegistryURL, f.CdiInstallNs)
 	tarArchiveURL := fmt.Sprintf(utils.TarArchiveURL, f.CdiInstallNs)
 	InvalidQcowImagesURL := fmt.Sprintf(utils.InvalidQcowImagesURL, f.CdiInstallNs)
+	externalCirrosURL := utils.InvalidQcowImagesURL
```
Comment (awels) [inline tests/datavolume_test.go]: Should this not be: CirrosExternalURL

Comment (j-griffith) [reply tests/datavolume_test.go]: Yes, it should; just noticed that in the failure thanks

**Comment by aglitke** [comment]
@j-griffith Have you created a bug against either qemu or fedora for the root cause?  We can hold back on Fedora for a little while but we should make sure to get the underlying issue resolved.

**Comment by mhenriks** [comment]
@aglitke the issue does not happen with qemu-img 4.0.0 I build locally

**Comment by aglitke** [comment]
Ok thanks @mhenriks .

File: hack/build/docker/builder/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM registry.fedoraproject.org/fedora-minimal:30
+FROM registry.fedoraproject.org/fedora-minimal:29
```
Comment (aglitke) [inline hack/build/docker/builder/Dockerfile]: Can we find a good place to comment why we are stuck on 29 so that we will remember later and can know when to upgarade to 31?

Comment (awels) [reply hack/build/docker/builder/Dockerfile]: We are adding a functional test that imports the image that is giving us trouble. During the course of normal development we will attempt to upgrade the images, and if grab a version that is broken it should fail that test.

File: hack/config-provider-external.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+docker_tag=devel
```
Comment (awels) [inline hack/config-provider-external.sh]: I don't think we want this in here.

File: test-pod.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+kind: Pod
```
Comment (awels) [inline test-pod.yaml]: I don't think this is needed is it?

**Comment by awels** [comment]
You also need to copy the file from the 'out' into the container when building. Add this to the Dockerfile of cdi-func-test-file-host-init
COPY cirros.qcow2 /tmp/source/cirros-qcow2.img

File: test-pod.yaml (hunk 5 above)
Comment (j-griffith) [reply test-pod.yaml]: No, it's not

File: hack/config-provider-external.sh (hunk 4 above)
Comment (j-griffith) [reply hack/config-provider-external.sh]: Nope, you don't

**Comment by j-griffith** [comment]
> Also need to edit tools/cdi-func-test-file-host-init/main.go to copy from /tmp/source to the actual target. Add something like this:
> 

@awels Ahh, thanks!  I think that's the thing that I was missing.

**Comment by awels** [comment]
@j-griffith there are some linting errors in here.

File: hack/build/docker/builder/fedora.repo [hunk 6]
```diff
@@ -1,4 +1,4 @@
-[fedora-base-fc30]
+[fedora-base-fc29]
```
Comment (awels) [inline hack/build/docker/builder/fedora.repo]: Need to do this for automation/check-patch.yumrepos as well. That is a symlink to all the other fedora.repos that are using in the dockerfiles.

Comment (j-griffith) [reply hack/build/docker/builder/fedora.repo]: I have that here no?  https://github.com/kubevirt/containerized-data-importer/pull/915/commits/52ad4935b02fb2a3f41d8aea41ce06cb4ffb0394#diff-42d85b9ddb8c0d2cec8713a2713f107cR1  Is there another hidden tgem

**Comment by j-griffith** [comment]
@awels Hmm... wonder if this fixes that too large test?  Causing it to now pass and thus fail?

**Comment by awels** [comment]
There that should work.

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
Trying one more time, but at least the k8s run worked last time.

**Comment by j-griffith** [comment]
ci test please

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
ci test please

**Comment by j-griffith** [comment]
ci test please

**Comment by awels** [comment]
ci test please
